### PR TITLE
Add more events in genuineCheck

### PR DESCRIPTION
**Pull request by gre**
This introduce more events to go through the genuine check that tells when the UI needs to display the allow manager.
It's a breaking change for genuine check usage so we'll HODL until we have the UI ready OR some code change to adapt with this new format.

### CODE REVIEW

**Comment by gre** [comment]
@valpinkman can you fix the checkDeviceForManager so it also returns GenuineCheckEvent ?

The next step for this rework:

- to backport the desktop code to make it work with the new type (trivial)
- to backport the changes on mobile as well without considering yet the UI changes
- to start thinking how to refactor mobile code to support is in the step system. This is probably not so trivial but there are maybe quick ways that can be found? There are basically 3 potential different rendering. We can unify the "before" and "after" into a same "spinner" for now but it worth asking for thoughts from design team.
- (bonus) we have the opportunity to have a better error when the user refuse that. That would mean introducing a new @ledgerhq/errors and throwing it in the observable in the case we have the specific status code at nonce=3

File: src/api/Manager.js [hunk 1]
```diff
@@ -273,16 +276,45 @@ const API = {
...
       })
-    }).pipe(
-      last(),
-      filter(o => o.type === "result"),
-      map(o => o.payload || "")
+    }).pipe(input =>
+      Observable.create(o => {
+        let timeout;
+        let requested;
+        input.subscribe({
```
Comment (gre) [inline src/api/Manager.js]: We need to return the subscription to support unsubscribe

File: demo/src/demos/transports/index.js [hunk 2]
```diff
@@ -142,6 +164,7 @@ class GenuineCheckButton extends Component<*, *> {
       )
     ).subscribe({
       next: result => {
+        console.log(result);
```
Comment (meriadec) [inline demo/src/demos/transports/index.js]: i mean

File: src/api/Manager.js (hunk 1 above)
Comment (valpinkman) [reply src/api/Manager.js]: Yes I saw that on documentation. I'll fix it

File: demo/src/demos/transports/index.js (hunk 2 above)
Comment (valpinkman) [reply demo/src/demos/transports/index.js]: it's a WIP, calm your tits ^^

File: src/api/Manager.js (hunk 1 above)
Comment (gre) [reply src/api/Manager.js]: How do we use the new GitHub wip feature?

File: demo/src/demos/transports/index.js (hunk 2 above)
Comment (valpinkman) [reply demo/src/demos/transports/index.js]: Also, it's the demo files

**Pull request merged**